polkadot/node/core/pvf/src/prepare/worker_intf.rs

```diff
@@ -196,14 +196,14 @@ async fn handle_response(
 	result: PrepareResult,
 	worker_pid: u32,
 	tmp_file: PathBuf,
-	artifact_path: PathBuf,
+	artifact_path_partial: PathBuf,
```

Comment:
I think your approach of returning the final artifact path in the success case makes sense. Just don't think we need to pass in a partial artifact here. If we instead passed the cache path, then we should have everything necessary to construct the artifact path at the point it's needed. Passing around a partial value seems wrong. For example, elsewhere in the file we are logging this value without indicating that it's a partial value, which can be confusing to readers of the log.

Reply:
This is the only place left that is not using the full path, coz here we don't have it yet. :)